File: src/main/java/nars/storage/NullBagObserver.java
package nars.storage;

import nars.entity.AbstractItem;

/**
 * a {@link BagObserver} that does nothing (null design pattern)
 */
public class NullBagObserver<BagType extends AbstractItem> implements BagObserver<BagType> {

    @Override
    public void setTitle() {
    }


    @Override
    public void setBag(Bag<BagType> concepts) {
    }


    @Override
    public void post(String str) {
    }


    @Override
    public void refresh(String string) {
    }


    @Override
    public void stop() {
    }
}

File: src/main/java/nars/storage/NullBagObserver.kt
package nars.storage

import nars.entity.ItemIdentity

/**
 * a [BagObserver] that does nothing (null design pattern)
 */
class NullBagObserver<BagType : ItemIdentity?> : BagObserver<BagType> {
    override fun setTitle() {}
    override fun setBag(concepts: Bag<BagType>?) {}
    override fun post(str: String?) {}
    override fun refresh(string: String?) {}
    override fun stop() {}
}